src/emqx_frame.erl

```diff
@@ -411,7 +411,7 @@ parse_reason_codes(Bin) ->
 
 parse_utf8_pair(<<Len1:16/big, Key:Len1/binary,
                   Len2:16/big, Val:Len2/binary, Rest/binary>>) ->
-    {{Key, Val}, Rest}.
+    {#{key => Key, value => Val}, Rest}.
 
```

Comment:
I think we should change this in emqx-rule-engine plugin.

Reply:
Thanks for your suggestion! I agree.

Fixed it in PR for rule-engine. This PR should be closed :)